src/LenderNft.hs

```diff
@@ -30,53 +30,47 @@ import qualified Ledger.Typed.Scripts     as Scripts
 import           Ledger.Value             as Value
 import qualified PlutusTx
 import           PlutusTx.Prelude         hiding (Semigroup (..), unless)
-import qualified Plutus.V1.Ledger.Scripts as Plutus
-import           Prelude                  (Semigroup (..), Show)
 import qualified Common.Utils             as U
 
-{-# INLINABLE lender #-}
-lender :: TokenName
-lender = TokenName { unTokenName = consByteString 76 emptyByteString }  -- L
-
 {-# INLINABLE mkPolicy #-}
-mkPolicy :: ValidatorHash -> TxOutRef -> Integer -> ScriptContext -> Bool
-mkPolicy vh utxo _ ctx = validate
+mkPolicy :: TxOutRef -> ScriptContext -> Bool
+mkPolicy utxo ctx = all validate mintedValue
   where
-    nftIsSentToCollateralSc :: Bool
-    nftIsSentToCollateralSc = traceIfFalse "minted lender nft is not sent to collateral smart contract" (valueOf (U.valueToSc vh ctx) (ownCurrencySymbol ctx) lender == 1)
+    mintFlattened :: [(CurrencySymbol, TokenName, Integer)]
+    mintFlattened = flattenValue $ txInfoMint (scriptContextTxInfo ctx)
+
+    mintedValue :: [(CurrencySymbol, TokenName, Integer)]
+    mintedValue = filter (\(cs, _tn, _n) -> cs == ownCurrencySymbol ctx) mintFlattened
+
+    calculateTokenNameHash :: BuiltinByteString
+    calculateTokenNameHash = sha2_256 (consByteString (txOutRefIdx utxo) ((getTxId . txOutRefId) utxo))
```

Comment:
`consByteString` expects `Word8` type as the first parameter. To guard against a possible overflow there, please assert that the `txOutRefIdx utxo < 256`

Reply:
Added validation

Reply:
You added validation on the other part. I meant validating that the `txOutRefIdx`, the number (**not** `txOutRefId`) is < 256. It needs to represent a single byte in the `consByteString` function, hence the limitation

Reply:
Fixed